### ISSUE TITLE
Escorts can't jump from Edusa to Postverta, at endgame

### ISSUE THREAD

**Issue by alextd** [endless-sky/endless-sky#6937]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Edusa->Postverta at the endgame (with SSil Vida active) is pretty much the only example of this bug in the game.

Escorts don't use the jump drive to go to a system with lower "jump range" - one that it wouldn't be able to jump back from. You, as the flagship, can jump there, but not back.

Escorts, once they get to a system with lower jump range, think they can jump away - but get into a confused state when they fail to actually do so, landing on a planet over and over.

Code discussion below.

### Steps to Reproduce

0. Load the save below (OR Activate Ssil Vida (so that systems have "jump range" 50))
1. Go to Edusa
2. Jump to Postverta
3. See escort not jumping directly, taking a long way around. Bonus: See escort's jump status as red while jumping out.

### Expected Behavior

Escorts jump along the path you take.

### Link to save file

[John Oliver~~testPostVertaJump.txt](https://github.com/endless-sky/endless-sky/files/8885366/John.Oliver.testPostVertaJump.txt)

### Operating System

Windows 10

### Game Source

Built from source

### Game Version

master

### Additional Information

I poked around the code for a simple fix. It highlighted a point of confusion. I suggest a big change. So seeking wisdom first.

# DistanceMap

The DistanceMap pathing system. For the Map Panel: it starts at the player and expand outwards, finding all paths. But when an 
escort(independent ship) is finding a path to a destination - DistanceMap starts at the destination and expands from there, stopping once it finds the ship. This is backwards from the behavior for the Map panel. 

So, when DistanceMap considers adding an edge from System A to B, the "jump range" of A is used. For the map panel, that's correct. But for escort pathfinding, the DistanceMap search is in the opposite direction of travel, so the "jump range" for A is the destination, not the source. So it doesn't consider the path from B to A, so the escort doesn't make a jump from A to B. It should check B's jump range there. Or... scrap that, just reverse the search pattern to work like the map panel does it.

(This probably has never been a problem since jump ranges were constant until recently)

# TL;DR

The pathing for ship->system should just be reversed to match how it works on the map.

The DistanceMap edges are assumed to be bidirectional. They're not anymore due to "jump range". It'd be confusing as heck to make it work bidirectionally. So let's keep it consistent: start at the source and find the destination. Don't know why it isn't like that now. 

Otherwise, there's a lot of places where the correct jump range would have to be checked, and confusingly differs because of how the object was constructed. And even if that stays not reversed, what if a system had a higher jump range? The path search away from it would be fine. But the search from the destination to the source to it wouldn't know to make the big jump to the system with a bigger jump range. There'd need to be some universal max jump range. So no, just search outwards from the starting point.

**Comment by alextd** [comment]
After writing and thinking even harder this is probably something I can actually fix easy. But... seeking wisdom nonetheless.

**Comment by warp-core** [comment]
#7377 fixes the issue in this specific circumstance, but escorts are still, in general, not able to jump to a system with too low a jump range even if their origin system has enough jummp range.

**Comment by warp-core** [comment]
It looks like this is a duplicate of #6639. Closing this and leaving that one open.
